pkg/cmd/registry/setting/set/set.go

```diff
@@ -0,0 +1,160 @@
+package set
+
+import (
+	"context"
+	"github.com/AlecAivazis/survey/v2"
+	"github.com/redhat-developer/app-services-cli/pkg/cmd/registry/registrycmdutil"
+	"github.com/redhat-developer/app-services-cli/pkg/cmd/registry/rule/rulecmdutil"
+	"github.com/redhat-developer/app-services-cli/pkg/core/cmdutil/flagutil"
+	"github.com/redhat-developer/app-services-cli/pkg/core/ioutil/icon"
+	"github.com/redhat-developer/app-services-cli/pkg/core/ioutil/iostreams"
+	"github.com/redhat-developer/app-services-cli/pkg/core/localize"
+	"github.com/redhat-developer/app-services-cli/pkg/core/logging"
+	"github.com/redhat-developer/app-services-cli/pkg/core/servicecontext"
+	"github.com/redhat-developer/app-services-cli/pkg/shared/factory"
+	registryinstanceclient "github.com/redhat-developer/app-services-sdk-go/registryinstance/apiv1internal/client"
+	"github.com/spf13/cobra"
+	"k8s.io/utils/strings/slices"
+
+	"github.com/redhat-developer/app-services-cli/pkg/shared/contextutil"
+)
+
+type options struct {
+	IO             *iostreams.IOStreams
+	Connection     factory.ConnectionFunc
+	Logger         logging.Logger
+	localizer      localize.Localizer
+	Context        context.Context
+	ServiceContext servicecontext.IContext
+
+	registryID     string
+	settingName    string
+	value          string
+	resetToDefault bool
+}
+
+// NewSetCommand creates a new command to set a service registry setting
+func NewSetCommand(f *factory.Factory) *cobra.Command {
+
+	opts := &options{
+		IO:             f.IOStreams,
+		Connection:     f.Connection,
+		Logger:         f.Logger,
+		localizer:      f.Localizer,
+		Context:        f.Context,
+		ServiceContext: f.ServiceContext,
+	}
+
+	cmd := &cobra.Command{
+		Use:     "set",
+		Short:   f.Localizer.MustLocalize("setting.set.cmd.description.short"),
+		Long:    f.Localizer.MustLocalize("setting.set.cmd.description.long"),
+		Example: f.Localizer.MustLocalize("setting.set.cmd.example"),
+		Args:    cobra.NoArgs,
+		RunE: func(cmd *cobra.Command, _ []string) (err error) {
+
+			var missingFlags []string
+
+			if opts.settingName == "" {
+				missingFlags = append(missingFlags, "setting-name")
+			}
+			if opts.value == "" && !opts.resetToDefault {
+				missingFlags = append(missingFlags, "value")
+			}
+
+			if !opts.IO.CanPrompt() && len(missingFlags) > 0 {
+				return flagutil.RequiredWhenNonInteractiveError(missingFlags...)
+			}
+
+			if len(missingFlags) > 0 {
+				err = runInteractivePrompt(opts, missingFlags)
```

Comment:
Very nice pattern to use interactive mode when flags are not set, I should use this more often 😄